FastSurferCNN/run_prediction.py

```diff
@@ -379,7 +379,10 @@ def check_and_download_ckpts(ckptpath):
                         default=os.path.join(os.path.dirname(__file__),
                                              "config/FastSurferVINN_sagittal.yaml"), type=str)
 
-    parser.add_argument('--no_cuda', action='store_true', default=False, help="Disables GPU usage")
+    parser.add_argument('--no_cuda', help="Deprecated: Use --device flag instead!")
+    parser.add_argument('--device', default="cuda",
```

Comment:
There should probably an "auto" option for device, which should be the default.

This is so the default configuration works independent of the hardware of the PC.

I think we also had some automatic identification of Hardware in fastsurfer before.

Reply:
I would even go as far as raising an exception, if cuda is passed as the device, but torch does not run on the GPU (e.g. the device is invalid)

Reply:
Similarly, batch_size should probably also default to some automatic behavior if no number is given to ensure the default configuration most likely works.

I am going to create a dedicated issue for this, since this is not really required for v2.0 and can be done in a later update.

Reply:
Currently, we just switch to cpu if gpu is set, but not available and return a warning. Assertion works as well. For auto, this only makes sense in my opinion once we include the other options (MAC M3 and what not). @m-reuter wanted to create a check list for that.

Reply:
My idea is that if you specifically specify cuda, you should get an error if it can't do it on the GPU.

This is a clear expected behavior that also lets you debug well. 

See also https://peps.python.org/pep-0020/

Reply:
Yeah, I am not arguing against it. I would just change it in one go with the auto option.

Reply:
Reply I updated it to "auto" so it will be easier in the future to add more options.